internal/net/httpserver.go

```diff
@@ -36,12 +42,15 @@ type HTTPServer struct {
 	lock     sync.Mutex
 	listener net.Listener
 	done     chan error
-	stopped  uint32
+	stopped  atomic.Bool
```

Comment:
Not sure to understand the point of this atomic. The `lock` above + `done` should be sufficient. I don't think the lock is such a hot path that it needs an atomic in front.